### ISSUE TITLE
Error when no ".env" file is present

### ISSUE THREAD

**Issue by devillex** [rolodato/dotenv-safe#11]
I get the following error when a `.env` file is not present, but the variables are still available in the environment:

```
{ Error: ENOENT: no such file or directory, open '.env'
    at Error (native)
    at Object.fs.openSync (fs.js:640:18)
    at Object.fs.readFileSync (fs.js:508:33)
    at Object.config (/pipeline/source/node_modules/dotenv/lib/main.js:30:37)
    at Object.config (/pipeline/source/node_modules/dotenv-safe/index.js:25:16)
    at Object.<anonymous> (/pipeline/source/webpack.config.js:1:86)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12) errno: -2, code: 'ENOENT', syscall: &#Hash: 02f39e618514ebe1e2c2
```

The error does not block the build of the project from proceeding (i.e. fail with exit code != 1), but it is a bit misleading.

Setup:
- Node v6.9.0
- npm v3.10.8
- macOS Sierra v10.12

**Comment by rolodato** [comment]
Hi @devillex, thanks for reporting this. You could argue that this is a bug or not, but it's certainly an unusual situation. Do you have some context about how you ran into this? Usually when you are using dotenv or its variants, the expectation is to always have a `.env` file somewhere in your project. I can use that information to decide how to go about fixing this. Thanks!

**Comment by devillex** [comment]
Hi @rolodato. Sure, thanks for the prompt response!

I am using it with a React app that uses Firebase. I use it specifically during the CI (via wercker) build step with webpack to replace env variables in code as part of the bundling process.

Here is my `webpack.config.js`:
```javascript
require('dotenv-safe').load(); // eslint-disable-line

const path = require('path');
const webpack = require('webpack');

module.exports = {
	entry: {
		app: [
			'babel-polyfill', // Adds ES6 built-ins like Promise and WeakMap
			'whatwg-fetch', // Adds 'fetch' polyfill
			path.join(__dirname, '/app/entry.js'), // app starts here
		],
	},
	output: {
		path: path.join(__dirname, '/build/js'),
		filename: '[name].bundle.js', // add this to the script tag
		publicPath: '/js/',
	},
	devtool: 'source-map',
	module: {
		loaders: [
			{ // pass js and jsx files through babel
				test: /\.jsx?$/,
				loader: 'babel-loader',
				exclude: /node_modules/,
				query: { presets: ['es2016', 'react', 'stage-0'] },
			},
			{ // import styles from 'styles.css'
				test: /\.css$/,
				loader: 'style!css?modules',
			},
		],
	},
	resolve: {
		root: path.resolve('./app'),
		extensions: ['', '.js', '.jsx'],
	},
	plugins: [
		new webpack.DefinePlugin({
			// this enables us to replace environment variables (see entry.js) in the code. The format is unfortunate,
			// but this is due to the string encoding
			'process.env': {
				// Firebase
				FIREBASE_API_KEY: `"${process.env.FIREBASE_API_KEY}"`,
				FIREBASE_AUTH_DOMAIN: `"${process.env.FIREBASE_AUTH_DOMAIN}"`,
				FIREBASE_DATABASE_URL: `"${process.env.FIREBASE_DATABASE_URL}"`,
				FIREBASE_STORAGE_BUCKET: `"${process.env.FIREBASE_STORAGE_BUCKET}"`,
			},
		}),
	],
};
```

**Comment by rolodato** [comment]
Thanks, that use case makes sense. It'll be a while before I can make some time to fix this, but if you're up for a PR + tests I can certainly look into that.

**Comment by ericf** [comment]
I ran into this as well. I think it would be helpful to display an error message about the missing `.env` file but one that isn't the standard Node `ENOENT` error. Also I think the errors would be more effective without the noise of the stack traces.

**Comment by rolodato** [comment]
Hi @ericf, for now I'm thinking that a missing `.env` file should not cause an ENOENT, so the ugly error message shouldn't be an issue. There will still be an error message when `.env.example` is not found which as of today is still a standard `ENOENT`, but we can take a look at that later.

**Comment by ericf** [comment]
When deploying my app, it doesn't contain an `.env` file, so it ends up logging the `ENOENT` error when that's actually the expected behavior since all of the environment variables already exist/exposed.

My use case is that `.env` is used for dev only. I ended up writing this script to help make less noisy error messages: https://github.com/ericf/timebox/blob/master/scripts/check-env but the `ENOENT` one still get logged and isn't captured by the `try/catch`.

**Comment by devillex** [comment]
@rolodato: I've submitted a PR that should resolve this issue. Please have a look when you get a chance.

**Comment by rolodato** [comment]
Fixed and released 3.0.0. Thanks everyone and @devillex for the PR!
